src/System.Memory/tests/Performance/Perf.ReadOnlySequence.Slice.cs

```diff
@@ -2,765 +2,338 @@
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            public Segments_100() : base(new ReadOnlySequenceFactory<byte>.SegmentsTestSequenceFactory(100)) { }
         }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_Memory_PosPos(int bufSize, int bufOffset)
-        {
-            var manager = new CustomMemoryForTest<byte>(new byte[bufSize], bufOffset, bufSize - 2 * bufOffset);
-            var buffer = new ReadOnlySequence<byte>(manager.Memory);
-            SequencePosition start = buffer.GetPosition(0);
-            SequencePosition end = buffer.GetPosition(0, buffer.End);
+    }
 
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(start, end);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
-        }
+    public abstract class Perf_ReadOnlySequence_Slice<T>
+    {
 
-        #endregion
+        private const int InnerCount = 10_000;
+        volatile static int _volatileInt = 0;
 
-        #region Byte_SingleSegment
+        internal ReadOnlySequenceFactory<T> Factory { get; }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_SingleSegment_Long(int bufSize, int bufOffset)
+        public Perf_ReadOnlySequence_Slice(ReadOnlySequenceFactory<T> factory)
         {
-            var segment1 = new BufferSegment<byte>(new byte[bufSize]);
-            var buffer = new ReadOnlySequence<byte>(segment1, bufOffset, segment1, bufSize - bufOffset);
-            long offset = buffer.Length / 10;
-
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<byte> temp = buffer.Slice(offset);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            Factory = factory;
         }
 
+        
         [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void Byte_SingleSegment_LongLong(int bufSize, int bufOffset)
+        [InlineData(10_000, 10)]
+        public void Long(int bufSize, int posCount)
```

Comment:
Can we rename the tests to be more clear rather than using Long, LongPos, etc.?
How about SliceStartOffset, SliceStartPosition, etc.?

Reply:
I think it would be nice.